### ISSUE TITLE
Packaging for Debian and derivatives (Ubuntu etc)

### ISSUE THREAD

**Issue by FedericoCeratto** [babluboy/bookworm#244]
Hello and thanks for developing Bookworm.
FYI packaging for Debian is ongoing at https://salsa.debian.org/debian/bookworm

**Comment by babluboy** [comment]
@FedericoCeratto Many thanks for your interest in Bookworm and your efforts in making it available cross distro.

Can you please update this issue when porting is completed for Debian and installation instructions.
I will add the Debian instructions to the Bookworm website.

**Comment by agharbeia** [comment]
There seems to be a problem with both of the packages available for Ubuntu, i.e:
bookworm 1.0.0-0~43~ubuntu0.4.1.1 from Xenial repository
 and
com.github.babluboy.bookworm1.1.2~0~2019082661223~ubuntu0.4.1.1 from ppa:bookworm-team/bookworm
Neither packages install any binaries or scripts, and both install only a doc directory with a couple of files.

**Comment by FedericoCeratto** [comment]
The packaging for Debian is blocked by #283

**Comment by babluboy** [comment]
> There seems to be a problem with both of the packages available for Ubuntu, i.e:
> bookworm 1.0.0-0~43~ubuntu0.4.1.1 from Xenial repository
> and
> com.github.babluboy.bookworm1.1.2~0~2019082661223~ubuntu0.4.1.1 from ppa:bookworm-team/bookworm
> Neither packages install any binaries or scripts, and both install only a doc directory with a couple of files.

I checked launchpad and the Xenial build for Bookworm looks fine and the package contains the source code and other debian packaging files. Can you please check the following links again.
[Xenial Package](https://code.launchpad.net/~bablu-boy/+archive/ubuntu/bookworm-daily/+sourcepub/10719789/+listing-archive-extra)
[Xenial Build Status for the Daily Receipe](https://code.launchpad.net/~bablu-boy/+archive/ubuntu/bookworm-daily/+recipebuild/2438243)
[Xenial build status for the Stable Receipe](url)

**Comment by babluboy** [comment]
> The packaging for Debian is blocked by #283

Just wanted to clarify, why the debian build failed - I understand the issue #283 needs to be fixed, but that cannot be the technical issue for why the debian build is failing at : https://salsa.debian.org/debian/bookworm

Looking at the error:
gbp:error: upstream/1.1.1 is not a valid treeish
ERROR: Job failed: exit code 1
I got the following explanation:
The current tag/branch you are in, is not a Debian source tree, it doesn't contain the debian/ directory in its root.

Is this something you are aware of to fix for the debian build or does it need me to update the source code to include any files in my debian directory. Or are you waiting for me to fix #283 so that you can make the build work for debian ?

**Comment by FedericoCeratto** [comment]
If you got the error form the CI pages salsa.debian.org - the CI is entirely ancillary and not part of the packaging process. The "real" package builds and installs files correctly.
I'm waiting for a solution for #283 to upload it into Debian once it's secure.
